pyccl/halos/profiles_2pt.py

```diff
@@ -185,5 +185,4 @@ def fourier_2pt(self, prof, cosmo, k, M, a,
             if not isinstance(prof2, HaloProfileCIBShang12):
```

Comment:
Since you created a superclass for CIB profiles, can we extend this to all CIB subclasses?

Reply:
See though, we are coming back to what I was telling you about the whole `Profile2pt` class holding instructions for a binary operation. Internally, this calls `prof._fourier_variance`, so to do what you are proposing we need every CIB's `_fourier_variance()` to have the same signature and to guarantee that the operation is commutative.

In which case, we can simply move `_fourier_variance` from CIBShang12 to its base, CIB. Otherwise this won't work at all.
It is a (kind of) big change, but I still think the best plan of action here is to treat 2-point correlation as binary operators and implement them in the individual profile.